### DISCUSSION TITLE
Error with generation and reading of types

### DISCUSSION THREAD

**Discussion by jukialen** [prisma/prisma#17495] [Q&A]
### Bug description

Hey,
I have 2 problems with types (after "migrate" and "generate").
I did migrations over again, but I still have these problems.
When I do POST I have:
"
PrismaClientValidationError:
Invalid this.prisma.friends.create() invocation in
/usr/src/app/src/friends/friends.service.ts:82:33
  79 addedFriend.length > 0 &&
  80   new NotAcceptableException(We're already friends.);
  81
→ 82 await this.prisma.friends.create({
       data: {
         username: '82277575-6ca3-4dc8-98b1-91bc8e8918fa',
         ~~~~~~~~
         friendId: c1428c3e-70cd-4f52-a8d2-d6022c96ab27,
         ~~~~~~~~
     +   friend: {
     +     create?: UsersCreateWithoutFriendInput | UsersUncheckedCreateWithoutFriendInput,
     +     connectOrCreate?: UsersCreateOrConnectWithoutFriendInput,
     +     connect?: UsersWhereUniqueInput
     +   },
     +   usersFriends: {
     +     create?: UsersCreateWithoutUserInFriendInput | UsersUncheckedCreateWithoutUserInFriendInput,
     +     connectOrCreate?: UsersCreateOrConnectWithoutUserInFriendInput,
     +     connect?: UsersWhereUniqueInput
     +   },
     ?   id?: String,
     ?   createAt?: DateTime,
     ?   updateAt?: DateTime
       }
     })

Unknown arg `username` in data.username for type FriendsCreateInput.
Unknown arg `friendId` in data.friendId for type FriendsCreateInput. Did you mean `friend`?
Argument friend for data.friend is missing.
Argument usersFriends for data.usersFriends is missing.

Note: Lines with + are required, lines with ? are optional.
Note: Lines with + are required, lines with ? are optional
" 
(in Insomnia I have 201 and "no body returned for response).

2) In Delete method I have:
"
src/friends/friends.controller.ts:145:53 - error TS2345: Argument of type '{ usernameId: string; }' is not assignable to parameter of type 'FriendsWhereUniqueInput'.
  Object literal may only specify known properties, and 'usernameId' does not exist in type 'FriendsWhereUniqueInput'.
145     return await this.friendsService.deleteFriend({ usernameId: username });
                                                        ~~~~~~~~~~~~~~~~~~~~
src/friends/friends.service.ts:101:13 - error TS2339: Property 'usernameId' does not exist on type 'FriendsWhereUniqueInput'.
101     const { usernameId } = where;
                ~~~~~~~~~~
".

### How to reproduce

In the case 1. error, I edited type in index.d.ts, but I think I shouldn't change anything in this file (then types FriendsUncheckedCreateInput and FriendsCreateInput are equal). Whereas in 2. case, I have good type in these file so I don't know what I would try to do and your extension also suggests the usernameId.

Functions in different tables I do in the same way (except Files where for POST I had to change type from FilesCreateInput to FilesUncheckedCreateInput but It helped. I don't understand why ) and it works good.

I did migrations over again, but I still have these problems.

### Expected behavior

I do POST and DELETE methods without error with types.

### Prisma information

<!-- Do not include your database credentials when sharing your Prisma schema! -->
```prisma
// Add your schema.prisma
https://pastebin.com/X20Z2Sdt
```

```ts
// Add your code using Prisma Client
https://pastebin.com/6qdntncp 
https://pastebin.com/vpwaiep1
https://pastebin.com/BYfaThhA
```


### Environment & setup

Fedora 37
Postgres
Node v19.3.0

### Prisma Version

```
4.9.0
```

**Comment by nurul3101** [comment]
Hello @jukialen 👋 

You mentioned that you edited the default generated types by Prisma, can you let us know what change was done in the generated types?

**Reply by jukialen** [reply]
usernameId and friendId in FriendsCreateInput.

**Reply by jukialen** [reply]
I cannot edit so 2 comment.
And friend and usersFriends in FriendsUncheckedCreateInput. 

It's already work. I  used latest among others Postgres, node (nestjs) and Docker desktop, I deleted node_modukes and dist folder, I run "docker system prune" and few times migrate and generate commands (first I deleted changes in these types in index.d.ts file.
I only already have problem with Patch method although I write the method and input data in the same way as in other models..

**Reply by nurul3101** [reply]
Thanks for getting back, so if I understand correctly POST and DELETE methods are now working as expected, but now only the PATCH method is not working as expected?

**Reply by jukialen** [reply]
Yes. It doesn't see data object xd In Groups module is ok, but in Friends althrough write a code and body data in the same way, no work.

`@Patch(':id')
async update(
  @Param('id') id: string,
  @Body('data')
  data: Prisma.FriendsUpdateInput | Prisma.FriendsUncheckedUpdateInput,
): Promise<Friends> {
  return this.friendsService.updateFriend({ where: { id }, data });
}

async updateFriend(params: {
  where: Prisma.FriendsWhereUniqueInput;
  data: Prisma.FriendsUpdateInput | Prisma.FriendsUncheckedUpdateInput;
}): Promise<Friends> {
  try {
    const { where, data } = params;
    console.log(data);
    console.log(where);
    await this.cacheManager.del(`friend ${where.id} `);
    return this.prisma.friends.update({ data, where });
  } catch (e) {
    console.error(e);
  }
}

@Patch(':name')
async updateGroup(
  @Param('name') name: string,
  @Body('data')
  data: Prisma.GroupsUpdateInput,
): Promise<GroupsModel> {
  return this.groupsService.updateGroup({ where: { name }, data });
}
async updateGroup(params: {
  where: Prisma.GroupsWhereUniqueInput;
  data: Prisma.GroupsUpdateInput;
}): Promise<Groups> {
  const { where, data } = params;
  return this.prisma.groups.update({ data, where });
}
`

{
	"data": {
		"favorite": true
	}
}

Error:
undefined
{ id: 'xyzxyzxyzxyz' }
[Nest] 48  - 01/25/2023, 9:40:08 AM   ERROR [ExceptionsHandler]
Invalid `this.prisma.friends.update()` invocation in
/usr/src/app/src/friends/friends.service.ts:104:34
  101 console.log(data);
  102 console.log(where);
  103 await this.cacheManager.del(`friend ${where.id} `);
→ 104 return this.prisma.friends.update({
      + data: FriendsUpdateInput | FriendsUncheckedUpdateInput,
        where: {
          id: '27afc977-e10b-48a1-b357-315a6d3568b2'
        }
      })
Argument data is missing.
Note: Lines with + are required

**Reply by nurul3101** [reply]
Is your project perhaps open source? If yes, could you share the link?

**Reply by jukialen** [reply]
Sorry for the late reply. I changed my schema littleso first I tried to fix. 
https://github.com/jukialen/pfartists_back

**Reply by nurul3101** [reply]
This is the [method](https://github.com/jukialen/pfartists_back/blob/main/backend/src/friends/friends.controller.ts#L131) that throws this error, right?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
